editors_checklist.md

```diff
@@ -0,0 +1,51 @@
+# Editors' checklist
+
+Upon submission:
```

Comment:
Maybe suggest that if there's issues we haven't dealt with yet, e.g,. pkg overlap, to open up a discussion at discuss.ropensci.org